pkg/query/query.go

```diff
@@ -291,29 +292,35 @@ func (s *Statement) Execute(ctx context.Context) (result *results.Result, err er
 		}
 	}
 
-	var rs = make(results.Rows, len(agg.aggregatedMap))
+	var rs = make(results.Rows, agg.aggregatedMap.Len())
 	count := 0
 
-	for key, val := range agg.aggregatedMap {
+	for i := agg.aggregatedMap.Iter(); i.Next(); {
 
-		if key.Time != 0 {
-			ts := time.Unix(key.Time, 0)
-			rs[count].Labels.Timestamp = &ts
+		key := types.ExtendedKey(i.Key())
+		val := i.Val()
+
+		if ts, hasTS := key.AttrTime(); hasTS {
+			rs[count].Labels.Timestamp = time.Unix(ts, 0)
 		}
-		rs[count].Labels.Iface = key.Iface
-		rs[count].Labels.HostID = key.HostID
-		rs[count].Labels.Hostname = key.Hostname
+		rs[count].Labels.Iface, _ = key.AttrIface()
+
+		// TODO: We need a nice solution for these special attributes - It's too expensive / complex
+		// to carry them around in the ExtendedKey struct across the board
+
+		// rs[count].Labels.HostID = key.AttrHostID()
```

Comment:
As discussed: these can be assigned "statically", since every query is executed in the scope of a single host.

Reply:
Jup, done!